utils/luci-app-ffwizard-berlin/luasrc/tools/freifunk/assistent/tools.lua

```diff
@@ -76,6 +76,24 @@ function firewall_find_zone(name)
 end
 
 
+-- checks if root-password has been set via CGI has_root-pass 
+function hasRootPass()
+	logger ("checking for root-password ...")
+
+	local isPasswordSet = true
+	local f = io.popen("wget http://localhost/cgi-bin/has_root-pass -q -O -")
+	local l = f:read("*a")
+	-- reply can be "password_is_set:no" / "password_is_set:no"
+	-- use sting:len() to check as this is more quick than substring() ot find()
```

Comment:
"Performance" is the only reason? It's not like we call this function 10,000 times per second. Checking the actual string will be less brittle and more transparent.